src/main/java/com/epam/training/designpatterns/fastfoodrestaurant/tables/Client.java

```diff
@@ -0,0 +1,78 @@
+package com.epam.training.designpatterns.fastfoodrestaurant.tables;
+
+import com.epam.training.designpatterns.fastfoodrestaurant.foods.Food;
+import com.epam.training.designpatterns.fastfoodrestaurant.foods.Menu;
+import com.epam.training.designpatterns.fastfoodrestaurant.staff.Server;
+
+import java.util.Random;
+import java.util.concurrent.Semaphore;
+
+/**
+ * Knows the server, and the menu
+ */
+public class Client implements Runnable{
+    private static int clientCount;
+    private final Semaphore sema;
+
+    private double happiness = 100;
+    private final int clientNumber;
+    private Food food;
+    private final int SIMULATION_SPEED;
+    private final Server server;
+
+    public Client(Server server, Semaphore sema, int simulation_speed){
+        SIMULATION_SPEED = simulation_speed;
+        this.clientNumber = clientCount++;
+        this.sema = sema;
+        this.server = server;
+    }
+    private void orderFood() throws InterruptedException {
+        Food food = Menu.randomFoodAndCondiment();
```

Comment:
Statikus metódus hívása nem szerencsés unit tesztelés szempontjából. Ki tudod ezt váltani? Mondjuk az étterem adhat egy étlapot a vendégnek, amikor belép. (Ott több dolog is lesz, pl. asztal elfoglalása = Semaphore.Acquire(), meg akkor egy étlap elvétele)